src/gc.rs

```diff
@@ -154,6 +157,35 @@ impl Collector {
         }
     }
 
+    fn enter_sweep_phase(&mut self) {
+        *self.state.lock().unwrap() = CollectorState::Sweeping;
+
+        for PtrInfo { ptr, size, gc } in AllocMetadata.iter() {
```

Comment:
I don't know what this does because `PtrInfo`'s fields aren't documented.

Reply:
I now suspect this might be faster if written as:

```rust
for PtrInfo { ptr, size, ... } in AllocMetadata.iter().filter(|x| x.gc) {
  ...
}
```

since the compiler is more likely to work out that it can optimise the iteration. I might be wrong though!

Reply:
Yeah I agree that's better. Fixed: 39f2f41538b00f52d58f183b2fe1d0f5526d3faa